src/modules/profile/index.ts

```diff
@@ -75,30 +83,64 @@ export async function fetchProfileWithConsistencyCheck(address: string): Promise
 }
 
 export async function redeployExistingProfile(
-  entity: Entity,
+  profile: Profile,
   connectedAccount: string,
   connectedAccountIdentity: AuthIdentity
 ): Promise<void> {
-  // Get all available catalyst servers for rotation
-  const environment = config.get('ENVIRONMENT')
-  const network = environment === 'development' ? 'sepolia' : 'mainnet'
-  const catalystServers = getCatalystServersFromCache(network)
+  const snapshotUrls = Object.entries(profile.avatars?.[0]?.avatar?.snapshots ?? {}).reduce(
+    (acc, [file, url]) => ({ ...acc, [file]: url }),
+    {} as Record<string, string>
+  )
 
-  // Prepare catalyst URLs for rotation, starting with the current PEER_URL
-  const PEER_URL = config.get('PEER_URL')
-  const catalystUrls = [
-    PEER_URL + '/content',
-    ...catalystServers.map(server => server.address + '/content').filter(url => url !== PEER_URL + '/content')
-  ]
+  const [bodyBuffer, faceBuffer] = await Promise.all([
+    fetch(snapshotUrls['body']).then(response => response.arrayBuffer()),
+    fetch(snapshotUrls['face256']).then(response => response.arrayBuffer())
+  ])
+
+  const files = createSnapshotFilesMap(bodyBuffer, faceBuffer)
+  const metadata = await buildProfileMetadata(profile, files)
 
-  const MAX_ATTEMPTS = Math.min(catalystUrls.length, 3) // Try up to 3 catalysts or all available
+  await redeployWithCatalystRotation(connectedAccount, connectedAccountIdentity, files, metadata)
```

Comment:
NIT:

In the future we could move this logic to catalyst-client so it exposes a function in LambdasClient called `deployEntityFromProfile` and let it handle all this work on that side